### DISCUSSION TITLE
`getStaticProps`'s `revalidate` should not control static data TTL and also `Cache-Control` headers

### DISCUSSION THREAD

**Discussion by schlosser** [vercel/next.js#28557] [Ideas]
### Describe the feature you'd like to request

Today, a page generated with `getStaticProps` that does not return the `revalidate` key [will send the cache control header](https://github.com/vercel/next.js/blob/51559f5c645c6ad8ba1bf2da74bac284015f887e/packages/next/server/send-payload.ts#L35) `s-maxage=31536000, stale-while-revalidate`, which means that a CDN will cache the HTML across deploys, causing bugs. I would like to be able to say "The data behind this SSG page does not need to be updated between deploys" while also ensuring that the CDN does not cache my HTML. 

### Describe the solution you'd like

Don't set `s-maxage=31536000, stale-while-revalidate` if `revalidate` is missing, only if it is actually `false`, or otherwise separate the `Cache-Control` header from the datasource TTL.

### Describe alternatives you've considered

The workaround I am using now is to set `revalidate: 1`, but that means that my data sources behind the SSG page are getting hit more than necessary.

**Comment by laketea** [comment]
seriously agree，the revalidate is just the behavior in nextjs ssg，this should not effect deploys.
i also meat this problem. and is hard to override the default value of cache-control
